lib/logstash/inputs/beats.rb

```diff
@@ -206,9 +210,19 @@ def invoke(connection, &block)
   # We are using a proxy queue supporting blocking with a timeout and
   # this thread take the element from one queue into another one.
   def start_buffer_broker
-    @threadpool.post do
-      while !stop?
-        @output_queue << @buffered_queue.pop_no_timeout
+    Thread.new do
+      LogStash::Util.set_thread_name("[beats-input]-buffered-queue-broker")
+
+      begin
+        while !stop?
+          @output_queue << @buffered_queue.take
+        end
+      rescue InterruptionException => e
+        # If we are shutting down without waiting the queue to unblock
+        # we will get an `InterruptionException` in that context we will not log it.
+        @logger.error("Beats input: bufferered queue exception", :exception => e) unless stop?
+      rescue => e
+        @logger.error("Beats input: unexpected exception", :exception => e) unless stop?
```

Comment:
have you considered not having "unless stop?" in this branch, as it is an unexpected exception?